packages/typespec-ts/src/utils/operationUtil.ts

```diff
@@ -546,3 +551,91 @@ export function parseItemName(paged: PagedResultMetadata): string | undefined {
   // TODO: support the nested item names
   return (paged.itemsSegments ?? [])[0];
 }
+
+export type ServiceOperation = SdkServiceMethod<SdkHttpOperation> & {
+  oriName?: string;
+};
+export function getMethodHierarchiesMap(
+  context: SdkContext,
+  client: SdkClientType<SdkServiceOperation>
```

Comment:
I don't see much value of it as 
1. it will introduce global variable and needs to be initialized at other places.
2. the getMethodHierarchiesMap is equal to what we use to have to iterate over client.operationGroups.operations. 
3. I don't see a case whether there's so many operations such that the iterate over the operations would become a bottleneck.